src/block.js

```diff
@@ -191,19 +207,25 @@ const createUnsafe = ({ bytes, cid, value: maybeValue, codec }) => {
 
   if (value === undefined) throw new Error('Missing required argument, must either provide "value" or "codec"')
 
-  return new Block({ cid, bytes, value })
+  return new Block({
+    // eslint-disable-next-line object-shorthand
+    cid: /** @type {CID<T, Code, Alg, V>} */ (cid),
+    bytes,
+    value
+  })
 }
 
 /**
- * @template T
- * @template {number} Code
- * @template {number} Algorithm
+ * @template {unknown} T - Logical type of the data encoded in the block
+ * @template {number} Code - multicodec code corresponding to codec used to encode the block
+ * @template {number} Alg - multicodec code corresponding to the hashing algorithm used in CID creation.
+ * @template {API.Version} V - CID version
  * @param {Object} options
- * @param {CID} options.cid
- * @param {ByteView<T>} options.bytes
- * @param {BlockDecoder<Code, T>} options.codec
- * @param {Hasher<Algorithm>} options.hasher
- * @returns {Promise<Block<T>>}
+ * @param {CID<T, Code, Alg, V>} options.cid
```

Comment:
now we have an inconsistency between `create` and `createUnsafe` with one accepting a `Link` and the other a `CID`; this probably should be a `Links`?

```suggestion
 * @param {API.Link<T, Code, Alg, V>} options.cid
```